### ISSUE TITLE
Récent Lightroom update denies the Heic format

### ISSUE THREAD

**Issue by mpdeboer08** [milch/LRExportHEIC#6]
After updating Lightroom to v13, the plugin doesn’t seem to be affecting the image anymore. The result is in jpg and bit depth shows as 8bit instead of the 10 bit format selected during export

**Comment by milch** [comment]
Hey, do you have any more information? What macOS version are you running? Can you try running the following command on your generated image? 

```
magick identify -verbose /path/to/your/image.heic
```

**Comment by milch** [comment]
I did some testing and found that Lightroom 13 starts providing an 8-bit JPEG file as source instead of a TIFF file. I'll have to do more investigation to see if this is a bug in Lightroom or if they changed the way you are supposed to request the source file

**Comment by milch** [comment]
Alright, I'm pretty sure this is a bug. Previously, when you requested an image format in the post processing plugin you would always get that format regardless of the format that the user selected. This matches the behavior described in the Lightroom Classic SDK Guide (version `Adobe Lightroom Classic 13.0 Software Development Kit (Build: "202309270914-5a1c6485")`): 

```
-- Optional: If you want to change the render settings for each photo 
-- before Lightroom renders it, write something like the following.
-- If not, omit the renditionOptions definition, and also omit
-- renditionOptions from the call to filterContext:rendition()
local renditionOptions = {
  filterSettings = function( renditionToSatisfy, exportSettings )
    exportSettings.LR_format = 'TIFF'
    return os.tmpname()
    -- ... if you wanted Lightroom to generate TIFF files
    -- and override the configured filename when providing
    -- input images to this post-process action.
    -- By doing so, you assume responsibility for creating
    -- the file type that was originally requested and placing it -- in the location that was originally requested in your
    -- filter loop below.
  end, 
}
```

It seems that in Lightroom 13.0 the image format provided is ALWAYS the file type selected in the `File Settings` export dialog. If I select `'Export To: Hard Drive` and set the file format to `TIFF` with 16-bit there, I can see that the plugin generates a high bit depth HEIC file correctly: 

```
$ magick identify -verbose ~/LRExport/export.tif
Image:
...
  Format: HEIC (High Efficiency Image Format)
  Mime type: image/heic
...
  Depth: 10/16-bit
  Channels: 3.0
  Channel depth:
    Red: 16-bit
    Green: 16-bit
    Blue: 16-bit
...
```

By the way, it seems that the HEIC API we've been using actually preserves all 16-bit from the input instead of "downgrading" to 10-bit bitdepth. That explains the large file size produced by this plugin - I'll have to look into how to actually limit it to 10 bits at some point.

Unfortunately this also means that this is a bug that needs to be fixed by Adobe, and I don't know what their timelines for this kind of thing are. I'm guessing stuff like this ends up very very very low on the list of priorities, so it might be a few versions before they fix it.

**Comment by mpdeboer08** [comment]
Hey Milch, first of all thanks so much for the effort you're putting into this. Really it should be on Adobe to add the Heif format to the list of export options but it's still lacking...

I do understand the issue or bug you're mentioning, and it does explain the larger file sizes I was seeing if the bit depth remains at 16 bits, though it is unused information if MacOS reads it as a 10-bit heif file.

With the added bug of Lightroom to output an 8-bit jpg file it makes it completely useless to try and use your plugin to create a heif output until this is addressed by adobe.

For now I'll be exporting as TIFF and converting to heif trough an automator workflow, however I hope I can revert back to your plugin soon as it is a huge time-saver.

I'll also send a bug notice to adobe to create awareness on this topic.

Thanks again,
Marc

**Comment by milch** [comment]
Thank you! The only developer support I found through developer.adobe.com was through the forums, so I created a topic here: https://community.adobe.com/t5/lightroom-classic-bugs/lightroom-classic-sdk-postprocessrenderedphotos-renditionoptions-ignored/idi-p/14275097

**Comment by milch** [comment]
Adobe just released LR Classic 13.1, which seems to have fixed this issue. I was able to confirm that both the intermediate file matches what the plugin requests, and that the output is in the HEIC format
